code/modules/mob/living/basic/boss/drake/drake.dm

```diff
@@ -94,29 +92,31 @@
 	RegisterSignal(src, COMSIG_SWOOP_INVULNERABILITY_STARTED, PROC_REF(swoop_invulnerability_started))
 	RegisterSignal(src, COMSIG_LAVA_ARENA_FAILED, PROC_REF(on_arena_fail))
 	AddElement(/datum/element/change_force_on_death, move_force = MOVE_FORCE_DEFAULT)
+	AddElement(/datum/element/footstep, FOOTSTEP_MOB_HEAVY)
+	AddComponent(/datum/component/sightrange_on_aggro, aggro_range = 18)
 
-/mob/living/simple_animal/hostile/megafauna/dragon/Destroy()
+/mob/living/basic/boss/dragon/Destroy()
 	fire_cone = null
 	meteors = null
 	mass_fire = null
 	lava_swoop = null
 	return ..()
 
-/mob/living/simple_animal/hostile/megafauna/dragon/OpenFire()
+/mob/living/basic/boss/dragon/RangedAttack(atom/target)
```

Comment:
This isn't what you want to hear but all of this should probably be in the AI controlller.
The hint is that you're turning all of this decision-making off when there's a client- that means what we are talking about here is "AI".